### PR TITLE
Wrong use of particle_signed_distances to compute the distance 

### PR DESCRIPTION
**Pull request by wacyyang**
This PR is to fix an error on the wrong use of `particle_signed_distances ` to compute the distance to the solid domain at the quadrature point on each exterior boundary edge.

- [x] 2D
- [x] 3D

### CODE REVIEW

File: proteus/mprans/RANS3PF2D.h [hunk 1]
```diff
@@ -4243,6 +4235,19 @@ namespace proteus
...
+                double distance_to_omega_solid(1e10);
+                if (use_ball_as_particle == 1)
+                {
+                  get_distance_to_ball(nParticles, ball_center, ball_radius, x_ext, y_ext, z_ext, distance_to_omega_solid);
+                }
+                else
+                {
+                  for (int i = 0; i < nParticles; i++)
+                  {
+                    double distance_to_i_th_solid = ebq_global_phi_solid[i * nElementBoundaries_owned * nQuadraturePoints_elementBoundary + ebN * nQuadraturePoints_elementBoundary + kb];
```
[the remark this answers is not in the document]
Comment (cekees) [reply proteus/mprans/RANS3PF2D.h]: No, I think you need to pass in and ebqe-like array and use the `ebNE_kb` index as its used above. You need something that is n_external_boundaries x nquadrature points for each particle distance.

File: proteus/mprans/RANS3PF2D.h [hunk 2]
```diff
@@ -2875,6 +2860,15 @@ namespace proteus
                                          x,y,z,
                                          phi_solid[eN_k]);
                 }
+                else
+                {
+                  phi_solid[eN_k] = 1e10;
+                  for (int i = 0; i < nParticles; i++)
+                  {
+                    double distance_to_i_th_solid = particle_signed_distances[i * nElements_global * nQuadraturePoints_element + eN * nQuadraturePoints_element + k];
```
[the remark this answers is not in the document]
Comment (cekees) [reply proteus/mprans/RANS3PF2D.h]: Here you need the eN_k index, modified for the number of particles.

File: proteus/mprans/RANS3PF2D.h [hunk 3]
```diff
@@ -5505,9 +5506,7 @@ namespace proteus
                                      vf[eN_k],
                                      phi[eN_k],
                                      &normal_phi[eN_k_nSpace],
-                                     nParticles,
-                                     nQuadraturePoints_global,
-                                     &particle_signed_distances[eN_k],
+                                     phi_solid[eN_k],//computed in getResidual
```
Comment (cekees) [inline proteus/mprans/RANS3PF2D.h]: Here, I would be consistent with using `distance_to_omega_solid` as the input variable to evaluateCoefficients.

File: proteus/mprans/RANS3PF2D.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (wacyyang) [reply proteus/mprans/RANS3PF2D.h]: I realize what you mean now. The distance to Omega_solid is computed in `postStep` and `attachModels`, so I do not have to loop over all particles. I am going to fix it now.

**Pull request merged**